packages/wattles/lib/src/queries/where_builder.dart

```diff
@@ -0,0 +1,49 @@
+// ignore_for_file: avoid_returning_this
+
+import 'package:wattles/wattles.dart';
+
+/// {@template where_builder}
+/// A builder for creating where statements.
+/// {@endtemplate}
+class WhereBuilder<T, V> {
+  /// {@macro where_builder}
+  WhereBuilder(this._queryable, this._whereStatement);
+
+  final List<WhereBuilder<T, dynamic>> _ands = [];
+
+  Where? _result;
+
+  final SchemaQueryable _queryable;
+
+  final V Function(T) _whereStatement;
+
+  /// Turn the where into a equal operator where statement.
+  WhereBuilder<T, V> equals(V value) {
+    if (_result != null) {
+      // TODO(wolfen): better error?
+      throw Exception('Where already built.');
```

Comment:
Can't this just be ignored instead? I mean, just do nothing.

Reply:
Depends on how we want to solve it. If we ignore it then someone could do:

```dart
..where((todo) => todo.id)
  .equals(1)
  .equals(2);
```

And then only the last one will be resolved. I'll have to see what would be more useful

Reply:
ideally this should be a compiler error if we return custom types rather than always returning a `WhereBuilder` we could scope the public API based on the operation being applied.

Reply:
Agreed, that would solve the problem. Added that to the roadmap!